src/wacks4shop/shift4shop.py

```diff
@@ -60,27 +61,40 @@ def request_all_products(self):
     def determine_sales(self, timestamp: Optional[str]) -> list[Sale]:
         """
         Query the Shift4Shop API for all orders since the given timestamp. Then transform these orders
-        into a list of Sale objects
+        into a list of Sale objects. Manually dedupe incomplete orders from all orders using
+        order id as the API doesn't support filtering by multiple order statuses.
         """
-        params: dict = {"orderstatus": ORDER_STATUSES}
+        params: dict = {}
         if timestamp:
             params["datestart"] = timestamp
-        response = self._request_orders(params)
+        all_orders_response = self._request_orders(params)
 
-        if not response.ok:
+        incomplete_orders_params: dict = {"orderstatus": 7}
+        if timestamp:
+            incomplete_orders_params["datestart"] = timestamp
+        incomplete_orders_response = self._request_orders(incomplete_orders_params)
+        incomplete_order_ids = {order["OrderID"] for order in incomplete_orders_response.json()}
+        completed_orders = [order for order in all_orders_response.json() if order["OrderID"] not in incomplete_order_ids]
```

Comment:
if we filter out the incomplete orders and write a new timestamp to db, we would potentially end up dropping the incomplete orders and never alert on them. need to think more about how to deal with this and under what condition is an order considered "incomplete".

Reply:
incomplete orders seem to be when a user adds item to cart. so I think the timestamp probably gets updated when the person actually checks out which makes this a nonproblem. see https://discord.com/channels/1029101311184076810/1029389916431450164/1058786287286489200